README.md

```diff
@@ -160,21 +143,109 @@ try {
+
+##### Will top-level `await` encourage developers to use `import()` unnecessarily, which is less optimizable?
+
+Many JavaScript developers are learning about `import()` specifically as a tool for code splitting. People are becoming aware of the relationship between bundling and multiple requests, and learning how to combine them for good application performance. Top-level `await` doesn't really change the calculus--using `import()` from a top-level `await` will have similar performance effects to using it from a function. As long as we can tie top-level `await`'s educational materials into the existing knowledge of that performance tradeoff, we hope to be able to avoid counterproductive increases in the use of `import()`.
+
+#### What exactly is blocked by a top-level `await`?
+
+When one module imports another one, the importing module will only start executing its module body once the dependency's body has finished executing. If the dependency reaches a top-level await, that will have to complete before the importing module's body starts executing.
+
+#### Why doesn't top-level `await` block the import of an adjacent module?
+
+If one module wants to declare itself dependent on another module, for the purposes of waiting for that other module to complete its top-level `await` statements before the module body executes, it can declare that other module as an import.
+
+In a case such as the following, the printed order will be `"X1"`, `"Y"`, `"X2"`, because importing one module "before" another does not create an implicit dependency.
+
+```mjs
+// x.mjs
+console.log("X1");
+await new Promise(r => setTimeout(r, 1000));
+console.log("X2");
+```
+
+```mjs
+// y.mjs
+console.log("Y");
+```
+
+```mjs
+// z.mjs
+import "./x.mjs";
+import "./y.mjs";
+```
 
-#### Halting Progress
+Dependencies are required to be explicitly noted in order to boost the potential for parallelism: Most setup work that will be blocking due to a top-level await (for example, all of the case studies above) can be done in parallel with other setup work from unrelated modules. When some of this work may be highly parallelizable (e.g., network fetches), it's important to get as many of these queued up close to the start of execution as possible.
 
-Variant A would halt progress in the module graph until resolved.
+#### What is guaranteed about code execution order?
 
-Variant B offers a unique approach to blocking, as it will not block siblings execution.
+Regardless of whether top-level `await` is used, modules always start running in the same post-order traversal established in ES2015: execution of module bodies starts with the deepest imports, in the order that the import statements for them are reached. After a top-level `await` is reached, control may be passed to start the next module in this traversal order.
```

Comment:
This still sounds a bit like Variant A, doesn't it? Considering
```js
// u.mjs
console.log("u1");
await new Promise(r => setTimeout(r, 1000));
console.log("u2");
```
```js
// x.mjs
import "u";
console.log("x");
```
```js
// y.mjs
console.log("y");
```
```js
// z.mjs
import "x";
import "y";
console.log("z");
```
Without TLA in `u.mjs`, we'd get u - x - y - z.
But with the TLA in `u.mjs`, I would expect to get u1 - y - u2 - x - z. Does `y` really need to wait for `x`, because of the established traversal?

Reply:
y doesn't need to wait for x. I agree with your expectations of ordering, and this text tries to explain them, alongside @guybedford's specification text for them in #33. Could you point a little more closely to what part of this text made you think otherwise?

Reply:
Thanks for the confirmation that my expectations aren't off.
I was confused by "*modules always start running in the same post-order traversal established in ES2015*", which seems to suggest that `y` would need to start running after `x`. The "*After a top-level `await` is reached, control may be passed to start the next module in this traversal order*" seemed to contradict that a bit, but I thought it would be geared towards the x-y-z example in the previous section ("block the import of an adjacent module"), and there `x` still *starts* before `y`.

Reply:
Maybe a few tiny changes could clear this up:

> Regardless of whether top-level `await` is used, modules *initially* start running in the same post-order traversal established in ES2015: execution of module bodies starts with the deepest imports, in the order that the import statements for them are reached. After a top-level `await` is reached, control *will* be passed to start the next *(first?) ready-to-run* module in this traversal order.

Reply:
Thanks for clarifying. What do you think of the version I just uploaded?

Reply:
I like it :-)